File: MusicShop.Model/Guitar.cs
using MusicShop.Model.BaseModel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicShop.Model
{
    public class Guitar : Product
    {

        public string? Pickups { get; set; }
        public string? PickupConfiguration { get; set; }
        public int? Frets { get; set; }
        public virtual GuitarType? GuitarType { get; set; }

    }
}

File: UI/musicshop_admin/lib/models/guitar/guitar.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:musicshop_admin/models/Abstract/product.dart';
import 'package:musicshop_admin/models/brand/brand.dart';
import 'package:musicshop_admin/models/guitar_type/guitar_type.dart';

part 'guitar.g.dart';

@JsonSerializable()
class Guitar {
  int? id;
  GuitarType? guitarType;
  String? pickups;
  String? pickupConfiguration;
  int? frets;
  String? productNumber;
  Brand? brand;
  String? model;
  double? price;
  String? description;
  DateTime? covariant;
  DateTime? updatedAt;
  String? image;

  Guitar();

  factory Guitar.fromJson(Map<String, dynamic> json) => _$GuitarFromJson(json);

  Map<String, dynamic> toJson() => _$GuitarToJson(this);
}
